ledger-service/db-backend/src/main/scala/com/digitalasset/http/dbbackend/Queries.scala

```diff
@@ -590,46 +595,17 @@ object Queries {
     (allTpids diff grouped.keySet).view.map((_, Map.empty[Party, Off])).toMap ++ grouped
   }
 
-  // invariant: each element x of result has `x.values.flatten.size <= size`
+  // invariant: each element x of result has `x.values.forall(_.forall(_.size <= size))`
   private[dbbackend] def chunkBySetSize[K, V](
       size: Int,
       groups: NonEmpty[Map[K, NonEmpty[Set[V]]]],
-  ): Vector[NonEmpty[Map[K, NonEmpty[Set[V]]]]] = {
+  ): NonEmpty[Map[K, NonEmpty[Vector[NonEmpty[Set[V]]]]]] = {
     assert(size > 0, s"chunk size must be positive, not $size")
```

Comment:
Although that's pre-existing, should this better be a `require` instead of `assert`?

Reply:
#15581